client/src/components/feeds/post/comments/new_comment/NewCommentBoxFooter.tsx

```diff
@@ -96,18 +98,25 @@ function NewCommentBoxFooter(props: NewCommentBoxFooterProps) {
   return (
     <div className="NewCommentBoxFooter">
       {props.anonymous ? (
-        <AiOutlineEyeInvisible
-          className="AnonymousIcon"
-          onClick={props.anonymousToggle}
-        />
+        <button className="AnonymousToggle" onClick={props.anonymousToggle}>
+          <AiOutlineEyeInvisible className="AnonymousIcon" />
+          <small>
+            Posting <strong>anonymously</strong>
+          </small>
+        </button>
       ) : (
-        <AiOutlineEye
-          className="AnonymousIcon"
+        <button
+          className="AnonymousToggle"
           onClick={() => {
             props.anonymousToggle();
             openAnonPopup();
           }}
-        />
+        >
+          <AiOutlineEye className="AnonymousIcon" />
+          <small>
+            Posting as <strong>{props.user?.name}</strong>
```

Comment:
The only issue with this right now is that for longer names, commenting 4 or 5 comments deep can result in the name wrapping across multiple lines. There could be multiple fixes, i.e truncating with an ellipsis, only displaying the given name, or saying "posting publicly" instead of "posting as `<name>`" Could be cool to do this with isMobile or dependent on how deep a comment is.  

![pr](https://user-images.githubusercontent.com/98923054/158517040-5a5063b8-398e-4c07-8c42-e8403fbab391.png)

Reply:
Good point! I think I might try to move the button above the the comment box on mobile. (Also, I’ve seen layout issues on my phone with the reaction buttons on deeply nested comments — perhaps that should be fixed too?)

Reply:
Hm, we weren't aware of this issue until just now. Are you talking about how attempting to react on deeply nested comments seems to shift the layout of other comments, as well as automatically close the reaction bar? We will certainly be looking into this. Do you have any intuition into why this is happening? The layout shift seems pretty clear (the div is for some reason gaining slightly more width, allowing words to shift into previously too small spaces), but the automatic closing is a bit more puzzling.

Reply:
I think this is what’s happening:

1. The user taps on the reaction buttons. Since the buttons have an on-hover listener, the browser moves the simulated cursor over them and fires the event handler
2. The additional reaction icons appear. Since they (+ the reply button) are wider than the available space, they break the layout by forcing the parent element to get wider.
3. Now that the comment element is wider, the browser re-flows the comment body. Since the comment body is now wider, the text takes up fewer lines.
4. Since the comment body is shorter, the reaction buttons move up.
5. Now that the virtual pointer is no longer over the buttons, they are hidden.

The best way to fix this IMO would be to fix the layout shift — probably by hiding the reply button on mobile when showing all the reaction buttons.

(but what if people have provided all types of reactions? Then you'd need a different layout I guess)